### DISCUSSION TITLE
Link Auth Providers to an Account

### DISCUSSION THREAD

**Discussion by zakaria-chahboun** [supabase/supabase#3098] [Questions] [answered]
Hi supabase community :zap: !
If my user already has an account and he want one day to logged in with an other provider (google, facebook ..)
How i can Link that account with provider(s)? *Just like [firebase link prvider](https://firebase.google.com/docs/auth/android/account-linking)*

And most important case is: If i have an *anonymous* user - how i can link his account with a provider (email/password, google account ...) later.

Thanks :heart: !

**Comment by inian** [comment, marked as answer]
Hi @zakaria-chahboun, user accounts from multiple login providers are automatically linked if the email remains the same across the third party providers. 

We do not implement anonymous logins yet. We have an issue for that [here](https://github.com/supabase/gotrue/issues/68)

**Reply by lancejpollard** [reply]
@inian you didn't answer the main part of the question:

> How i can Link that account with provider(s)?

If they have different email addresses for each account (google is foo@gmail.com, github is bar@gmail.com, apple is baz@gmail.com), how can they be linked?
